libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationMetricsWrapper.java

```diff
@@ -105,8 +108,9 @@ static void rerouteEvent(SessionState sessionState, RouteProgress routeProgress,
       routeProgress.currentLegProgress().currentStep().duration().intValue(),
       (int) routeProgress.currentLegProgress().currentStepProgress().distanceRemaining(),
       (int) routeProgress.currentLegProgress().currentStepProgress().durationRemaining(),
-      sessionState.currentStepCount(), sessionState.originalStepCount()
-    ));
+      sessionState.currentStepCount(), sessionState.originalStepCount());
+    rerouteEvent.put(MapboxNavigationEvent.KEY_CREATED, TelemetryUtils.generateCreateDate(location));
```

Comment:
@cammace why are we taking the date from the location only here (reroute case)? I mean, why aren't we including the same line in the rest of the cases (i.e. arrive, cancel, depart and feedback events)? If we should do that, best option would be taking care of that in [MAS](https://github.com/mapbox/mapbox-java/blob/2dfe04b21a95cccf81e77329babf829ed6ce27a3/mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/navigation/MapboxNavigationEvent.java#L283).

cc @ericrwolfe @zugaldia

Reply:
@Guardiola31337 I think that what we learned from https://github.com/mapbox/mapbox-navigation-android/pull/398 (which is definitely a temporary fix) is that telemetry should provide a default value for `created` if none is provided, but allow the consuming SDK to manually set a value too.